applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -269,24 +281,31 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN)) {
-		/* Preferred min is 0, so we set min supported */
-		if (pref_dly_min == 0) {
+		if (pref_dly_min == 0 || pref_dly_min < pres_dly_min) {
 			*pres_dly_us = pres_dly_min;
-		} else {
+		} else if (pref_dly_min <= pres_dly_max) {
 			*pres_dly_us = pref_dly_min;
+		} else {
+			LOG_ERR("Min pref delay, %d, greater than max pres delay, %d", pref_dly_min,
+				pres_dly_max);
+
+			return -ENOTSUP;
 		}
 
 		return 0;
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MAX)) {
-		/* Preferred max is 0, so we set max supported */
-		if (pref_dly_max == 0) {
+		if (pref_dly_max == 0 || pref_dly_max > pres_dly_max) {
 			*pres_dly_us = pres_dly_max;
-		} else {
+		} else if (pref_dly_max < pres_dly_min) {
```

Comment:
I think this should be:
```suggestion
		} else if (pref_dly_max >= pres_dly_min) {
```
As it stands now the pres_dly_us would be less than pres_dly_min

Reply:
Thanks this is a pain to test as I have to change a number of Kconfigs to exercise each issue. Will workout how to test this!